#!/usr/bin/env python
PACKAGE = "screw_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("resizing_factor",   double_t, 0, "A float parameter to make the original image's resolution",  0.3) 
gen.add("screw_detection_confidence_threshold", double_t, 0, "Threshold for the predictions. Increase to have less false positives", 0.8) 
gen.add("torx6_radius_threshold", int_t, 0,"radius threshold to differentiate between torx 6 and torx 8", 5)
gen.add("hough_upper_threshold", int_t, 0, "One of the Hough parameters", 100)
gen.add("hough_lower_threshold", int_t, 0, "One of the Hough parameters", 50)
gen.add("hough_min_radius", int_t, 0, "One of the Hough parameters", 5)
gen.add("hough_max_radius", int_t, 0, "One of the Hough parameters", 30)
gen.add("mode", bool_t, 0, "Mode identifier: 0 for data collection, 1 for detection", True)

exit(gen.generate(PACKAGE, "screw_detection", "ScrewDetection"))
